hypertools/_shared/helpers.py

```diff
@@ -136,6 +136,44 @@ def reshape_data(x,labels):
         x_reshaped[categories.index(point)].append(x_stacked[idx])
     return [np.vstack(i) for i in x_reshaped]
 
+def is_label_probabilistic(labels):
+    """
+        Tells Whether the Assigned Label is a Scalar or Not[Case for GaussianMixture models]
+    """
+    if type(labels[0]) is np.ndarray or type(labels[0]) is list:
+        return True
+    return False
+def reshape_labels(labels):
+    """
+        Since, Most of the functionalities in Hypertool work with Scalar(To be Precise, Categorical in nature) Value of Label.
+        This method assigns the label/cluster to which the data point(s) most probably belong.  
+    """
+    if labels==None or len(labels)<1:
+        return labels
+    return [label.argmax() for label in labels]
+
+def get_plt_cluster_colors(n_clusters):
+    fig,ax= plt.subplots()#This Line needs a Fix: It assumes that the color palette is default. Ignores Users Input, if any.
+    color_cycle=ax._get_lines.prop_cycler
+    cluster_colors=range(n_clusters)
+    for i,color in enumerate(color_cycle):
+        if i<n_clusters:
+            cluster_colors[i]=color["color"]
+        else:
+            break
+    return np.array(cluster_colors)
+
+def transform_labels2RGB(n_clusters,cluster_discrete_labels,cluster_labels):
```

Comment:
how about something like:

```
def labels2RGB(cluster_colors, cluster_labels):
    """
    Map cluster labels to a weighted average of cluster_colors where the weights are defined by the label 
    probabilities
    """
    return list(map(lambda x, c=cluster_colors: np.dot(x, c)/np.sum(x), cluster_labels))
```

Reply:
Makes code better - Incorporated. 
However, last time I missed a  check which has been handled now:
The missing case was the scenario where all data points could be clustered using number of clusters lesser than n_clusters. It's not something which occurs frequently, but still a valid possibility.